scripts/zephyr_module.py

```diff
@@ -0,0 +1,145 @@
+# the build system.
+type: map
+mapping:
+  build:
+    required: true
+    type: map
+    mapping:
+      cmake:
+        required: false
+        type: str
+      kconfig:
+        required: false
+        type: str
+'''
+
+AUTO_GENERATED_HEADER = '# NOTE: THIS FILE IS AUTOGENERATED - DO NOT EDIT\n\n'
+
+
+def process_module(module, section_name, cmake_out=None, kconfig_out=None):
+    cmake_path = 'zephyr'
+    kconfig_file = 'zephyr/Kconfig'
+
+    module_yml = os.path.join(module, 'zephyr/module.yml')
+    if os.path.isfile(module_yml):
+        with open(module_yml, 'r') as f:
+            meta = yaml.safe_load(f.read())
+
+        schema = yaml.safe_load(METADATA_SCHEMA)
+        pykwalify.core.Core(source_data=meta, schema_data=schema).validate()
+
+        section = meta.get(section_name, dict())
+        if section is not None:
+            cmake_path = section.get('cmake', cmake_path)
+            kconfig_file = section.get('kconfig', kconfig_file)
+
+    cmake_path_abs = os.path.abspath(os.path.join(module, cmake_path))
+    cmake_file_abs = os.path.join(cmake_path_abs, 'CMakeLists.txt')
+    if os.path.isfile(cmake_file_abs) and cmake_out is not None:
+        val_str = '{}:{}\n'.format(os.path.basename(module), cmake_path_abs)
+
+        with open(cmake_out, 'a') as f:
+            f.write(val_str)
+
+    kconfig_file_abs = os.path.abspath(os.path.join(module, kconfig_file))
+    if os.path.isfile(kconfig_file_abs) and kconfig_out is not None:
+        val_str = 'osource "{}"\n\n'.format(kconfig_file_abs)
+
+        with open(kconfig_out, 'a') as f:
+            f.write(val_str)
+
+
+def generate_file_header(includefile):
+    path = os.path.dirname(includefile)
+    if path != '' and not os.path.exists(path):
+        os.makedirs(path)
+    if includefile is not None:
+        with open(includefile, 'w+') as f:
+            f.write(AUTO_GENERATED_HEADER)
+
+
+def main():
+    parser = argparse.ArgumentParser(description='''
+    Process a list of projects and create Kconfig / CMake include files for \
+    projects which are also a Zephyr module''')
+
+    parser.add_argument('--kconfig-out', required=False,
+                        help='File to write with resulting KConfig import'
+                             'statements.')
+    parser.add_argument('--cmake-out', required=False,
+                        help='File to write with resulting <name>:<path>'
+                             'values to use for including in CMake')
+    parser.add_argument('-s', '--section', required=False,
+                        help='Section in YAML file to parse')
+    parser.add_argument('-m', '--modules', nargs='+',
+                        help='List of modules to parse instead of using `west'
+                             'list`')
+    parser.add_argument('-x', '--extra-modules', nargs='+',
+                        help='List of extra modules to parse')
+    args = parser.parse_args()
+
+    if args.kconfig_out is not None:
```

Comment:
Just

if args.kconfig_out:

is best-practice.

Reply:
I don't think that is best practice, also PEP8 states:
https://www.python.org/dev/peps/pep-0008/#id51
> Comparisons to singletons like None should always be done with is or is not, never the equality operators.
> 
> Also, beware of writing if x when you really mean if x is not None

And here, I do indeed mean `is not None`

Reply:
> And here, I do indeed mean is not None.

Maybe you do, but we want to express "x is not None and x != empty_string, which we will get with "if x", but we don't get with a None comparison.

Allowing x == empty_string will cause weird issues further down the line AFAICT.

Reply:
@mbolivar : I agree, we wouldn't want to use "if x" for the int type.

Reply:
@SebastianBoe you're right, haven't consider that a user could write `--cmake-out=` without providing a filename in which case the value would indeed be an empty string.

So that is fixed now.